ext/node/polyfills/_zlib.mjs

```diff
@@ -155,10 +155,27 @@ export const inflateRawSync = function (buffer, opts) {
   return zlibBufferSync(new InflateRaw(opts), buffer);
 };
 
+function sanitizeInput(input) {
+  if (typeof input === "string") input = Buffer.from(input);
+
+  if (
+    !Buffer.isBuffer(input) &&
+    (input.buffer && !input.buffer.constructor === ArrayBuffer)
+  ) throw new TypeError("Not a string, buffer or dataview");
+
+  if (input.buffer) {
+    input = new Uint8Array(input.buffer);
+  }
```

Comment:
```suggestion
  if (input.buffer) {
    input = new Uint8Array(input.buffer);
  }
```

I'm not sure if this is the reason for test failure but the input view should respect offsets.

```js
input = new Uint8Array(input.buffer, input.byteOffset, input.byteLength)
```

In the test, the compressed buffer is truncated and passed to decompress(). It expects decompress() to fail.
https://github.com/denoland/deno/blob/59cfe20f3be727096d328ecb2196478ae01447f6/cli/tests/node_compat/test/parallel/test-zlib-truncated.js#L48-L51